Game will not start after update 12-6-20 (v2.0.0-alpha1)

I am running Android 11 on a Pixel 2XL.  This morning I had an update for Lexica and ran it.  After updating, I am unable to start the game in the application.

The application launches and I am able to access the menus for the various settings.  If I select "New Game" the screen blanks and returns to the initial page with "New Game", "About", "Restore Game", "Preferences".  If I persist in trying to start a game I get an Android system message telling me "Lexica keeps stopping" and gives me "App Info" and "Close app" options.  Selecting "Restore Game" has no effect or sign of actually doing anything.  I can select it as many times as I want and get no Android system message.

I have restarted the phone with no change to the behavior.   

I uninstalled Lexica and reinstalled it with no change to the behavior.

Oh gosh, thanks so much for the quick feedback everyone. I can confirm that this is related to the following crash:

```
         AndroidRuntime  E  FATAL EXCEPTION: main
                         E  Process: com.serwylo.lexica, PID: 18155
                         E  java.lang.RuntimeException: Unable to resume activity {com.serwylo.lexica/com.serwylo.lexica.GameActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'com.serwy
                            lo.lexica.game.Game$GameStatus com.serwylo.lexica.game.Game.getStatus()' on a null object reference
                         E      at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4902)
                         E      at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4945)
                         E      at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:52)
                         E      at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:190)
                         E      at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:105)
                         E      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2473)
                         E      at android.os.Handler.dispatchMessage(Handler.java:110)
                         E      at android.os.Looper.loop(Looper.java:219)
                         E      at android.app.ActivityThread.main(ActivityThread.java:8347)
                         E      at java.lang.reflect.Method.invoke(Native Method)
                         E      at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
                         E      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1055)
                         E  Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'com.serwylo.lexica.game.Game$GameStatus com.serwylo.lexica.game.Game.getStatus()' on a null object referen
                            ce
                         E      at com.serwylo.lexica.GameActivity.onResume(GameActivity.java:206)
                         E      at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1472)
                         E      at android.app.Activity.performResume(Activity.java:8222)
                         E      at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4892)
                         E      ... 11 more
```

I'll see if I can get quick fix out and release an `-alpha2`.

I also realise it is bad that you have all likely updated because F-Droid told you to, even though it is an Alpha release. That is also my fault, I've incorrectly misconfigured the F-Droid metadata such that all versions become the "suggested version". I'll leave it like that until this bug is fixed (so that new users and others updating don't get this bad bug), but after that, I'll try to fix the metadata so alpha builds are not thrust upon you!

Initial investigation shows that a local debug build from `git` works as expected, however the F-Droid build fails with the following earlier error:

```
                         E  android.content.res.Resources$NotFoundException: Resource ID #0x0
                         E      at android.content.res.ResourcesImpl.getValue(ResourcesImpl.java:328)
                         E      at android.content.res.ResourcesImpl.openRawResource(ResourcesImpl.java:467)
                         E      at android.content.res.Resources.openRawResource(Resources.java:1418)
                         E      at android.content.res.Resources.openRawResource(Resources.java:1362)
                         E      at com.serwylo.lexica.game.Game.initializeDictionary(Game.java:270)
                         E      at com.serwylo.lexica.game.Game.initializeDictionary(Game.java:263)
                         E      at com.serwylo.lexica.game.Game.setBoard(Game.java:235)
                         E      at com.serwylo.lexica.game.Game.<init>(Game.java:184)
                         E      at com.serwylo.lexica.GameActivity.newGame(GameActivity.java:124)
                         E      at com.serwylo.lexica.GameActivity.onCreate(GameActivity.java:91)
                         E      at android.app.Activity.performCreate(Activity.java:8085)
                         E      at android.app.Activity.performCreate(Activity.java:8073)
                         E      at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1320)
                         E      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3868)
                         E      at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4074)
                         E      at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:91)
                         E      at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:149)
                         E      at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:103)
                         E      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2473)
                         E      at android.os.Handler.dispatchMessage(Handler.java:110)
                         E      at android.os.Looper.loop(Looper.java:219)
                         E      at android.app.ActivityThread.main(ActivityThread.java:8347)
                         E      at java.lang.reflect.Method.invoke(Native Method)
                         E      at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
                         E      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1055)
```

This happens during `onCreate`, and implies that the release build doesn't end up with `res/raw/dictionary_LANG` files :(

I will have to look into why these are not present.

The good news is that it si reproducible locally if I run the F-Droid build here. It also means that I should add to my test plan 'Quick regression test on the F-Droid build prior to tagging for release".

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -10,8 +10,8 @@ android {
         applicationId "com.serwylo.lexica"
         minSdkVersion 18
         targetSdkVersion 28
-        versionCode 20000
-        versionName "2.0.0-alpha1"
+        versionCode 20001
+        versionName "2.0.0-alpha2"
 
         testInstrumentationRunner 'androidx.test.runner.AndroidJUnitRunner'
     }

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -92,9 +92,11 @@ dependencies {
 }
 
 afterEvaluate {
-    assembleDebug.dependsOn rootProject.buildDictionaries
-    if (tasks.findByPath('assembleRelease')) {
-        assembleRelease.dependsOn rootProject.buildDictionaries
+    generateDebugResources.dependsOn rootProject.buildDictionaries
+    generateDebugResources.shouldRunAfter rootProject.buildDictionaries
+    if (tasks.findByPath('generateReleaseResources')) {
+        generateReleaseResources.dependsOn rootProject.buildDictionaries
+        generateReleaseResources.shouldRunAfter rootProject.buildDictionaries
     }
 }
 repositories {

--- a/app/src/main/java/com/serwylo/lexica/GameActivity.java
+++ b/app/src/main/java/com/serwylo/lexica/GameActivity.java
@@ -201,6 +201,7 @@ public class GameActivity extends AppCompatActivity implements Synchronizer.Fina
         if (game == null) {
             Toast.makeText(this, R.string.error_restoring_game, Toast.LENGTH_SHORT).show();;
             NavUtils.navigateUpFromSameTask(this);
+            return;
         }
 
         switch (game.getStatus()) {

--- a/fastlane/metadata/android/en-US/changelogs/20001.txt
+++ b/fastlane/metadata/android/en-US/changelogs/20001.txt
@@ -1,0 +1,16 @@
+Game Modes!
+
+These preferences are replaced with "Game Modes":
+ * Duration
+ * Board size
+ * Min word length
+ * Scoring type
+ * Hint modes
+
+You can change game mode from the main screen, or create your own. High scores are based on game mode and lexicon.
+
+Fixed critical bug in 2.0.0-alpha1 (thanks to all who reported the issue).
+
+Please provide feedback/report issues at https://github.com/lexica/lexica/issues.
+
+Support Lexica development via Liberapay or GitHub Sponsors :)

